File: projects/m1/040-password-validation/solutions/alessiacapoeaso/password-validation.js
// check password and return true or false.
/* Good password to be a one that is:
 - at least 8 characters long
 - at least one uppercase letter
 - at least one lowercase letter
 - at least one number
*/
function checkPassword(password) {
  let characterUpperCase = 0;
  let characterLowerCase = 0;
  let numberCase = 0;
  for (let i = 0; i < password.length; i++) {
    if (parseInt(password[i]) >= 0 && parseInt(password[i]) <= 9) {
      numberCase += 1;
    } else if (password[i] === password[i].toUpperCase()) {
      characterUpperCase += 1;
    } else if (password[i] === password[i].toLowerCase()) {
      characterLowerCase += 1;
    }
  }
  if (password.length >= 8 && characterUpperCase >= 1 && characterLowerCase >= 1 && numberCase >= 1) {
    return true;
  } else {
    return false;
  }
}
const userPassword = prompt('Please enter your password')

if(checkPassword(userPassword)){
    alert('Valid password')
} else {
    alert('Invalid password')
}

File: projects/m1/040-password-validation/solutions/Jurgens Marpali/password_validation.py
def psw_validation(psw):
    val = True
    if len(psw) < 8:
        print("YOur password should be at least 8 character long!")
        val = False

    if not any(char.isdigit() for char in psw):
        print("YOur password should have at least one number")
        val = False
    
    if not any(char.isupper() for char in psw):
        print("Your passowrd should have at least one uppercase letter")
        val = False
    
    if not any(char.islower() for char in psw):
        print("Your passowrd should have at least one lowercase letter")
        val = False
    if val:
        return val

def main():
    psw = input("Write your password here:")
    if (psw_validation(psw)):
         print("Valid Password")
    else:
        print("Invalid password")

if __name__ in "__main__":
    main()